prov/util/src/util_attr.c

```diff
@@ -1017,13 +1026,18 @@ static void fi_alter_domain_attr(struct fi_domain_attr *attr,
 				 const struct fi_domain_attr *hints,
 				 uint64_t info_caps, uint32_t api_version)
 {
-	ofi_mr_mode_adjust(info_caps, &attr->mr_mode);
-
-	if (FI_VERSION_LT(api_version, FI_VERSION(1, 5)))
-		attr->mr_mode = attr->mr_mode & OFI_MR_BASIC_MAP ?
-			FI_MR_BASIC : FI_MR_SCALABLE;
-	else
-		attr->mr_mode &= ~(FI_MR_BASIC | FI_MR_SCALABLE);
+	int hints_mr_mode;
+
+	hints_mr_mode = hints ? hints->mr_mode : 0;
+	if (hints_mr_mode & (FI_MR_BASIC | FI_MR_SCALABLE)) {
+		attr->mr_mode = hints_mr_mode;
+	} else if (FI_VERSION_LT(api_version, FI_VERSION(1, 5))) {
+		attr->mr_mode = (attr->mr_mode && attr->mr_mode != FI_MR_SCALABLE) ?
+				FI_MR_BASIC : FI_MR_SCALABLE;
```

Comment:
This conversion converts a mr_mode specified using 1.5 semantics to 1.0 semantics.  Since the util providers pass the version through, I don't think we ever fall through this case.  It would only be used with a 1.0 app talking to a util provider, which uses 1.5 semantics with a core provider.